server/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetadata.java

```diff
@@ -378,7 +379,9 @@ private static void toInnerXContent(IndexTemplateMetadata indexTemplateMetadata,
             indexTemplateMetadata.settings().toXContent(builder, params);
             builder.endObject();
 
-            includeTypeName &= (params.paramAsBoolean("reduce_mappings", false) == false);
+            if(builder.getRestApiVersion() != RestApiVersion.V_7) {
```

Comment:
can this be onOrAfter V_8 ? (minor preference to avoid using 'not a specific version')